File: Hw03/practice/corner/nms.m
function [cx, cy, cs] = nms(cornerScore)
% This code is part of:
%
%   CMPSCI 370: Computer Vision, Spring 2016
%   University of Massachusetts, Amherst
%   Instructor: Subhransu Maji
%
%   Homework 3

% Pick local maxima
locs = imregionalmax(cornerScore, 8);

% Shrink connected components to one point
locs = bwmorph(locs, 'shrink', Inf);

% Zero out everything else
cornerScore = cornerScore.*locs;

% Convert binary image to a sorted list of locations
locinds = find(locs);
[cy, cx] = ind2sub(size(locs), locinds);
cs = cornerScore(locinds);
[~,ord] = sort(cs, 'descend');
cs = cs(ord);
cx = cx(ord);
cy = cy(ord);

File: Hw04/hw_practice01/code/nms.py
#This code is part of:
#
#  CMPSCI 370: Computer Vision, Spring 2018
#  University of Massachusetts, Amherst
#  Instructor: Subhransu Maji
#
#  Homework 3

import numpy as np
from scipy.ndimage.filters import maximum_filter
from skimage.morphology import local_maxima
from skimage.feature import peak_local_max

def nms(corner_score):

    #Pick local maxima in a window
    locs = peak_local_max(corner_score, 5)
    #Get scores at maximum peaks
    scores = corner_score[locs[:, 0], locs[:, 1]]
    #Sort scores
    idx_sorted = np.argsort(scores)
    #from IPython import embed; embed(); exit(-1)

    #Get scores and coordinates according to sorting
    cs = scores[idx_sorted]
    cx = locs[idx_sorted, 1]
    cy = locs[idx_sorted, 0]

    return cx, cy, cs
